### ISSUE TITLE
Put OpenShift initial user credentials for HTPasswd identity provider in `openshift-config` namespace

### ISSUE THREAD

**Issue by tolusha** [eclipse-che/che#19380]
### Is your task related to a problem? Please describe.
When Eclipse Che is deployed on a fresh OpenShift cluster then `openshift-oauth-user-credentials` secret is created in `eclipse-che` namespace. It contains OpenShift initial user credentials for HTPasswd identity provider.
The problem is the following. When Eclipse Che is deleted then the secret is deleted as well and credentials are lost.

### Describe the solution you'd like
Create a secret in `openshift-config` namespace.

**Comment by l0rd** [comment]
What about deleting the user when deleting Che?

**Comment by tolusha** [comment]
We don't delete this user intentionally since it is OpenShift cluster user not a Eclipse Che one.
But it is possible to delete it by setting `spec.auth.initialOpenShiftOAuthUser: false`
```
 initialOpenShiftOAuthUser:
                  description: For operating with the OpenShift OAuth authentication,
                    create a new user account since the kubeadmin can not be used.
                    If the value is true, then a new OpenShift OAuth user will be
                    created for the HTPasswd identity provider. If the value is false
                    and the user has already been created, then it will be removed.
                    If value is an empty, then do nothing. The user's credentials
                    are stored in the `openshift-oauth-user-credentials` secret by
                    Operator. Note that this solution is Openshift 4 platform-specific.
```

**Comment by l0rd** [comment]
@tolusha first of all I think this issue makes a lot of sense: last week I have deployed, deleted and redeployed Che. As a consequence I didn't had noted down the password, the secret got deleted and I had to create a new OpenShift user to be able to use Che.

That said, I think that as admin I would appreciate if the Che does a good cleanup at deletion. That means that we should remove the OpenShift user created at installation time. Creating the user in the `openshift-config` and not cleaning it up is a clever workaround. But that may be annoying for administrators to see that Che leaves some objects around after installation.

If `chectl server:deploy` creates a user, I would expect that `chectl server:delete` removes it. And the fact that the default user, created at startup, get deleted if the admin changes `initialOpenShiftOAuthUser: false` is not good UX in my opinion. But that's another story.
